### ISSUE TITLE
Undefined symbols for architecture arm64

### ISSUE THREAD

**Issue by RS360build** [BranchMetrics/unity-branch-deep-linking-attribution#123]
iOS Architectures set to "arm64 armv7 armv7s", Build Active Architecture Only to No.

Please, help.

Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_BranchUnityWrapper", referenced from:
      objc-class-ref in BranchAppController.o
  "__redeemRewardsForBucket", referenced from:
      _Branch_redeemRewards_m3000292141 in Bulk_Assembly-CSharp_0.o
      _Branch__redeemRewardsForBucket_m2570124600 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__redeemRewardsForBucket_m2570124600)
  "__getCreditsForBucket", referenced from:
      _Branch_getCredits_m3547202777 in Bulk_Assembly-CSharp_0.o
      _Branch__getCreditsForBucket_m1714161847 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getCreditsForBucket_m1714161847)
  "__getCredits", referenced from:
      _Branch_getCredits_m500603266 in Bulk_Assembly-CSharp_0.o
      _Branch__getCredits_m1490662785 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getCreditsForBucket_m1714161847, _Branch__getCredits_m1490662785 )
  "__getCreditHistoryForTransactionWithLengthOrderAndCallback", referenced from:
      _Branch__getCreditHistoryForTransactionWithLengthOrderAndCallback_m1711720893 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getCreditHistoryForTransactionWithLengthOrderAndCallback_m1711720893)
  "__getCreditHistoryForBucketWithCallback", referenced from:
      _Branch__getCreditHistoryForBucketWithCallback_m370174333 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getCreditHistoryForBucketWithCallback_m370174333)
  "__loadRewardsWithCallback", referenced from:
      _Branch_loadRewards_m3396010975 in Bulk_Assembly-CSharp_0.o
      _Branch__loadRewardsWithCallback_m1446088861 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__loadRewardsWithCallback_m1446088861)
  "__accountForFacebookSDKPreventingAppLaunch", referenced from:
      _Branch_accountForFacebookSDKPreventingAppLaunch_m2574732210 in Bulk_Assembly-CSharp_0.o
      _Branch__accountForFacebookSDKPreventingAppLaunch_m4269451093 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__accountForFacebookSDKPreventingAppLaunch_m4269451093)
  "__listOnSpotlight", referenced from:
      _Branch_listOnSpotlight_m3989893032 in Bulk_Assembly-CSharp_0.o
      _Branch__listOnSpotlight_m2814397350 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__listOnSpotlight_m2814397350)
  "__registerView", referenced from:
      _Branch_registerView_m3915808218 in Bulk_Assembly-CSharp_0.o
      _Branch__registerView_m2306933985 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__registerView_m2306933985)
  "__setMaxRetries", referenced from:
      _Branch_setMaxRetries_m301002430 in Bulk_Assembly-CSharp_0.o
      _Branch__setMaxRetries_m1453011322 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setMaxRetries_m1453011322)
  "__userCompletedActionWithState", referenced from:
      _Branch__userCompletedActionWithState_m326743976 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__userCompletedActionWithState_m326743976)
  "__initSessionWithUniversalObjectCallback", referenced from:
      _Branch_initSession_m909057453 in Bulk_Assembly-CSharp_0.o
      _Branch__initSessionWithUniversalObjectCallback_m648502127 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__initSessionWithUniversalObjectCallback_m648502127)
  "__setRetryInterval", referenced from:
      _Branch_setRetryInterval_m1649790864 in Bulk_Assembly-CSharp_0.o
      _Branch__setRetryInterval_m1108862872 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setRetryInterval_m1108862872)
  "__setDebug", referenced from:
      _Branch_setDebug_m694632402 in Bulk_Assembly-CSharp_0.o
      _Branch__setDebug_m2385939096 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setDebug_m2385939096)
  "__redeemRewards", referenced from:
      _Branch_redeemRewards_m476443017 in Bulk_Assembly-CSharp_0.o
      _Branch__redeemRewards_m2870731437 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__redeemRewardsForBucket_m2570124600, _Branch__redeemRewards_m2870731437 )
  "__logout", referenced from:
      _Branch_logout_m844728253 in Bulk_Assembly-CSharp_0.o
      _Branch__logout_m3739877891 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__logout_m3739877891)
  "__setRequestMetadata", referenced from:
      _Branch__setRequestMetadata_m4253635193 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setRequestMetadata_m4253635193)
  "__setBranchKey", referenced from:
      _Branch_Awake_m3007394441 in Bulk_Assembly-CSharp_0.o
      _Branch__setBranchKey_m3260480113 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setBranchKey_m3260480113)
  "__initSessionAsReferrableWithCallback", referenced from:
      _Branch_initSession_m1287863262 in Bulk_Assembly-CSharp_0.o
      _Branch__initSessionAsReferrableWithCallback_m1397015380 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__initSessionAsReferrableWithCallback_m1397015380)
  "__getLatestReferringBranchLinkProperties", referenced from:
      _Branch_getLatestReferringBranchLinkProperties_m3506500526 in Bulk_Assembly-CSharp_0.o
      _Branch__getLatestReferringBranchLinkProperties_m3452811640 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getLatestReferringBranchLinkProperties_m3452811640)
  "__setIdentity", referenced from:
      _Branch_setIdentity_m3207608447 in Bulk_Assembly-CSharp_0.o
      _Branch__setIdentity_m2519274146 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setIdentityWithCallback_m3012121641, _Branch__setIdentity_m2519274146 )
  "__initSessionWithCallback", referenced from:
      _Branch_initSession_m3538034369 in Bulk_Assembly-CSharp_0.o
      _Branch__initSessionWithCallback_m3792061888 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__initSessionWithCallback_m3792061888)
  "__sendEvent", referenced from:
      _Branch_sendEvent_m778628739 in Bulk_Assembly-CSharp_0.o
      _Branch__sendEvent_m2411677812 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__sendEvent_m2411677812)
  "__resetUserSession", referenced from:
      _Branch_resetUserSession_m1841138536 in Bulk_Assembly-CSharp_0.o
      _Branch__resetUserSession_m4248580179 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__resetUserSession_m4248580179)
  "__getFirstReferringBranchLinkProperties", referenced from:
      _Branch_getFirstReferringBranchLinkProperties_m3694989779 in Bulk_Assembly-CSharp_0.o
      _Branch__getFirstReferringBranchLinkProperties_m4130690916 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getFirstReferringBranchLinkProperties_m4130690916)
  "__getCreditHistoryForBucketWithTransactionLengthOrderAndCallback", referenced from:
      _Branch__getCreditHistoryForBucketWithTransactionLengthOrderAndCallback_m1855150287 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getCreditHistoryForBucketWithTransactionLengthOrderAndCallback_m1855150287)
  "__getFirstReferringBranchUniversalObject", referenced from:
      _Branch_getFirstReferringBranchUniversalObject_m3683043749 in Bulk_Assembly-CSharp_0.o
      _Branch__getFirstReferringBranchUniversalObject_m1169413175 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getFirstReferringBranchUniversalObject_m1169413175)
  "__shareLinkWithLinkProperties", referenced from:
      _Branch__shareLinkWithLinkProperties_m3575078740 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__shareLinkWithLinkProperties_m3575078740)
  "__userCompletedAction", referenced from:
      _Branch_userCompletedAction_m2606252603 in Bulk_Assembly-CSharp_0.o
      _Branch__userCompletedAction_m400854117 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__userCompletedAction_m400854117, _Branch__userCompletedActionWithState_m326743976 )
  "__getLatestReferringBranchUniversalObject", referenced from:
      _Branch_getLatestReferringBranchUniversalObject_m1913314127 in Bulk_Assembly-CSharp_0.o
      _Branch__getLatestReferringBranchUniversalObject_m3384599739 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getLatestReferringBranchUniversalObject_m3384599739)
  "__setNetworkTimeout", referenced from:
      _Branch_setNetworkTimeout_m348740200 in Bulk_Assembly-CSharp_0.o
      _Branch__setNetworkTimeout_m3125223926 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setNetworkTimeout_m3125223926)
  "__getCreditHistoryWithCallback", referenced from:
      _Branch_getCreditHistory_m1187275898 in Bulk_Assembly-CSharp_0.o
      _Branch__getCreditHistoryWithCallback_m4282881690 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getCreditHistoryWithCallback_m4282881690)
  "__setIdentityWithCallback", referenced from:
      _Branch__setIdentityWithCallback_m3012121641 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__setIdentityWithCallback_m3012121641)
  "__getShortURLWithBranchUniversalObjectAndCallback", referenced from:
      _Branch__getShortURLWithBranchUniversalObjectAndCallback_m4279279474 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__getShortURLWithBranchUniversalObjectAndCallback_m4279279474)
  "__initSessionAsReferrable", referenced from:
      _Branch_initSession_m2322767046 in Bulk_Assembly-CSharp_0.o
      _Branch__initSessionAsReferrable_m2314414307 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__initSessionAsReferrableWithCallback_m1397015380, _Branch__initSessionAsReferrable_m2314414307 )
  "__initSession", referenced from:
      _Branch_initSession_m2906224887 in Bulk_Assembly-CSharp_0.o
      _Branch__initSession_m3813292091 in Bulk_Assembly-CSharp_0.o
     (maybe you meant: _Branch__initSessionWithCallback_m3792061888, _Branch__initSessionAsReferrableWithCallback_m1397015380 , _Branch__initSession_m3813292091 , _Branch__initSessionAsReferrable_m2314414307 , _Branch__initSessionWithUniversalObjectCallback_m648502127 )
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by dimixar** [comment]
Have the same error here. Using Unity 2017.2.0p4 with latest unity branch SDK.

**Comment by dimixar** [comment]
Setting architecture to armv7 didn't help

**Comment by antonargunov** [comment]
@dimixar 
1) check please you have .m files in your xcode project
2) if you don't have, open Unity and check files in ios plugin (Plugins/Branch/IOS/Branch-SDK/Branch-SDK), files must be flagged for "ios platfrom". If files isn't flagged - they will not be added into xcode project.

**Comment by dimixar** [comment]
Checked, the .m files are there. They are flagged with iOS platform only. And they persist in XCode project.

**Comment by dimixar** [comment]
Found the answer on our issue.
It so happens that the source code on GitHub and the source code from the unity package provided by documentation are different somehow.

Basically, we removed the code from unity project and installed again the unity package provided by the documentation.

The only inconvenience was that after generating the XCode project, we had to enable manually the "Associated Domains" capability and add our domain links. It would be awesome if you'd add this as a PostProcess Build script that would modify the PBX project with the necessary capabilities enabled.
